### DISCUSSION TITLE
Invalid CSS Error

### DISCUSSION THREAD

**Discussion by Isaac-Tait** [jekyll/jekyll#9028] [Q&A] [answered]
I have a very basic app that I am attempting to install [Bootstrap](https://getbootstrap.com/) into. However, I am getting this error when I run `jekyll serve`:
```
Error: Invalid CSS after "-": expected 1 selector or at-rule, was "---" (Jekyll::Converters::Scss::SyntaxError)
        on line 1:1 of assets/css/main.scss
        from line 1:1 of main.scss
>> ---

   ^
```

The file where the error is cropping up is `main.scss`. Here are the entirety of its contents:

```
---
---

@import "main";
@import 'bootstrap/bootstrap';
```

Note: I have tried deleting `@import "main";` and the error just shifts down to the remaining `@import` statement...

`_config.yml` 

```
sass:
  sass_dir: ../assets/css

url: "" # the base hostname & protocol for your site, e.g. http://example.com
baseurl: "" # the subpath of your site, e.g. /blog
title: "" # the name of your site, e.g. ACME Corp.
```

My `main.scss` file is throwing this error in VSCode as well `{ expectedscss(css-lcurlyexpected)`

I was following [this](https://github.com/maltyeva/jekyll-bootstrap-starter) tutorial and when I clone her repo I get the same error... [Here](https://github.com/Isaac-Tait/Jekyll) is my repo just in case it is helpful. 

I also setup a `GitHub Code Space` [here](https://isaac-tait-jekyll-w9rvpwv6c9g4j.github.dev/). Just run `bundle` then `jekyll serve` and you should get **both** errors...

**Comment by Isaac-Tait** [comment]
Possibly related to https://github.com/jekyll/jekyll/issues/7816 ? At least the error messages are similar...

**Comment by pranabdas** [comment, marked as answer]
Hello @Isaac-Tait, Can you try using `bundler`? In case you don't have it installed, you can follow the installation instructions <https://jekyllrb.com/docs/installation/> If I run `bundle install` followed by `bundle exec jekyll serve` in your project, I do not see any error. However, your CSS linking is incorrect. Your `main.css` is generated at `/assets/main.css` (as it is processed from `assets/main.scss`), but you link `/assets/css/main.css` in `_layouts/default.html`. You can either link `/assets/main.css` or place your `main.scss` at `assets/css/main.scss`.

**Comment by Isaac-Tait** [comment]
Thank you @pranabdas for this. That is odd about the bundler, I had followed the `jekyll` docs on installation (the only difference being that I used `rbnv` instead because it was already setup/installed beforehand). Once I fisxed the `_layouts/default.html` file though I was able to get things running. Thanks again!
